tests/testAutopas/tests/containers/TraversalComparison.cpp

```diff
@@ -0,0 +1,236 @@
+#include "autopasTools/generators/RandomGenerator.h"
+
+/**
+ * Generates a random 3d shift with the given magnitude. The shift is uniformly distributed on a sphere with radius
+ * magnitude.
+ * @param magnitude
+ * @param generator
+ * @return
+ */
+std::array<double, 3> randomShift(double magnitude, std::mt19937 &generator) {
+  std::uniform_real_distribution<double> uniform01(0.0, 1.0);
+  double theta = 2 * M_PI * uniform01(generator);
+  double phi = acos(1 - 2 * uniform01(generator));
+  double x = sin(phi) * cos(theta) * magnitude;
+  double y = sin(phi) * sin(theta) * magnitude;
+  double z = cos(phi) * magnitude;
+  return {x, y, z};
+}
+
+/**
+ * Adds shifts with the given magnitude to all particles.
+ * The shifts are generated in order of the particle id and with a fixed seed to ensure a reproducible behavior.
+ * @tparam ContainerPtrType
+ * @param containerPtr
+ * @param magnitude
+ * @param totalNumParticles
+ */
+template <class ContainerPtrType>
+void TraversalComparison::executeSlightShift(ContainerPtrType containerPtr, double magnitude,
+                                             unsigned long totalNumParticles) {
+  std::vector<std::array<double, 3>> shiftVectorByID(totalNumParticles);
+  unsigned seed = 42;
+  std::mt19937 generator(seed);
+  for (auto &elem : shiftVectorByID) {
+    elem = randomShift(magnitude, generator);
+  }
+  unsigned long numIteratedParticles = 0;
+  for (auto &mol : *containerPtr) {
+    mol.addR(shiftVectorByID[mol.getID()]);
+    ++numIteratedParticles;
+  }
+  EXPECT_EQ(numIteratedParticles, totalNumParticles);
+}
+
+/**
+ * Calculates the forces for a given configuration.
+ * @param containerOption Specifies the container.
+ * @param traversalOption Specifies the traversal.
+ * @param dataLayoutOption Specifies the data layout.
+ * @param newton3Option Specifies whether the newton3 optimization should be used or not.
+ * @param numMolecules The number of molecules.
+ * @param numHaloMolecules The number of halo molecules.
+ * @param boxMax The maximum of the simulation box. The minimum is {0.,0.,0.}
+ * @param cellSizeFactor The cell size factor.
+ * @param doSlightShift Specifies whether to add random shifts of size skin/2 to all particles after the neighbor list
+ * generation.
+ * @return Tuple of forces for all particles, ordered by particle id, and array of the globals values (upot, double).
+ */
+std::tuple<std::vector<std::array<double, 3>>, std::array<double, 2>> TraversalComparison::calculateForces(
+    autopas::ContainerOption containerOption, autopas::TraversalOption traversalOption,
+    autopas::DataLayoutOption dataLayoutOption, autopas::Newton3Option newton3Option, unsigned long numMolecules,
+    unsigned long numHaloMolecules, std::array<double, 3> boxMax, double cellSizeFactor, bool doSlightShift) {
+  // Construct container
+  autopas::ContainerSelector<Molecule, FMCell> selector{_boxMin, boxMax, _cutoff};
+  double skin = _cutoff * 0.1;
```

Comment:
```suggestion
  constexpr double skin = _cutoff * 0.1;
```
since `_cutoff` is also `constexpr